### PR TITLE
ROX-19877: Add crypto util packages

### PR DESCRIPTION
**Pull request by charmik-redhat**
## Description

Added util package to get the key from volume mount.
Added crypto package to perform encryption and decryption.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Unit tests

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: pkg/crypto/aes_gcm.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+// EncryptAESGCM performs AES-GCM encryption on the given text and returns the encrypted
+// bytes as a base64 standard encoded string. The encryption key should be a base64 standard encoded string.
+func EncryptAESGCM(keyString string, stringToEncrypt string) (string, error) {
```
Comment (clickboo) [inline pkg/crypto/aes_gcm.go:16]: Might be good to leave this funcs as Encrypt and Decryptand build an interface for it, and have this as the AES GCM implementation of that interface. Later if we choose another encryption standard, we implement another class for it.

File: central/notifiers/utils/encryption.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package utils
+
+import (
+	"os"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	encryptionKeyFile = "/run/secrets/stackrox.io/notifiers/enc-key"
```
Comment (theencee) [inline central/notifiers/utils/encryption.go:10]: @johannes94 where are you actually going to mounting the key to?

Comment (johannes94) [reply central/notifiers/utils/encryption.go:10]: According to this [PR](https://github.com/stackrox/stackrox/pull/7889/files) by Vlad it'll be `/run/secrets/stackrox.io/central-encryption-key` the current data field is `encryptionKey` so the full path to the file would be `/run/secrets/stackrox.io/central-encryption-key/encryption-key`.

If we want a different default path we can still change it easily in Vlad's PR.

Comment (charmik-redhat) [reply central/notifiers/utils/encryption.go:10]: This is fine. I can change the path in my PR.

Comment (charmik-redhat) [reply central/notifiers/utils/encryption.go:10]: Also @johannes94 , is the key stored as a plain string or as a base64 std/url encoded string?

**Review by clickboo — APPROVED** [review]
lgtm, but needs a make gen to pass style ebfore merge.

File: central/notifiers/utils/encryption.go (hunk 2 above)
Comment (charmik-redhat) [reply central/notifiers/utils/encryption.go]: Resolved on slack.

**Pull request merged**